src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -878,7 +878,7 @@ protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
             {
                 var info = new DataViewSchema.DetachedColumn[_parent._outputs.Length];
                 info[0] = new DataViewSchema.DetachedColumn(_parent._outputs[0], new VectorDataViewType(NumberDataViewType.Single, _parent._classCount), null);
-                info[1] = new DataViewSchema.DetachedColumn(_parent._outputs[1], NumberDataViewType.UInt32, null);
+                info[1] = new DataViewSchema.DetachedColumn(_parent._outputs[1], new KeyDataViewType(typeof(uint), _parent._classCount), ((DataViewSchema.Column)InputSchema.GetColumnOrNull(_parent._labelColumnName)).Annotations);
```

Comment:
>_outputs [](start = 68, length = 8)

Please replace this field with two fields containing the score column name and the predicted label column name, and do the same for the argument in the `Options` class.
It is implicitly assumed throughout this code that there are exactly two output columns, the first one being the score column and the second one being the predicted label column - for example, this method (`GetOutputColumnsCore`), `GetOutputSchema` in the estimator, `GetOutputColumnsCore` in the mapper.

Reply:
I just noticed that there is the same problem with the input column names - on one hand the argument in the `Options` class is an array, and in other place it is assumed that it contains exactly one element which is the features column.

---
In reply to: [326619105](https://github.com/dotnet/machinelearning/pull/4228#discussion_r326619105) [](ancestors = 326619105)